File: Arrays/MajorityElement/Python/169-Majority-Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count_map = {}

        for num in nums:
            count_map[num] = count_map.get(num, 0) + 1

        maj_cnt = len(nums) // 2


        for k, v in count_map.items():
            if v > maj_cnt:
                return k

File: Arrays/MajorityElement/Go/169-Majority-Element.go
func majorityElement(nums []int) int {
    count_map := make(map[int]int)

    for _, num := range nums {
        count_map[num] ++
    }

    maj_ele := len(nums) / 2

    for k, v := range count_map {
        if v > maj_ele {
            return k
        }
    }
    return -1
}